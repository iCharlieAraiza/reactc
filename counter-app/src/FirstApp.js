import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({greeting, name, mockHandler})=>{
    
    const numbers = [1,2,3,4,5,6,7];
 
    const user = {
        name: "Carlos",
        lastName: "Surname",
        age: 27
    }

    const clickHandle = ()=>{
        console.log("hola");
        mockHandler();
    }

    return(
        <>
            <h1> { greeting } </h1>
            <p>Este es un ejemplo de prueba para hacer un componente con</p>
            <ul>
                <li> { numbers } </li>
            </ul>
            <pre>
                <p>Nombre: { name }</p>
                <p>Apellido: { user.lastName }</p>
                <p>Edad: { user.age } </p>
            </pre>
            <button onClick={clickHandle}>prueba de click</button>

        </>
        )
}

FirstApp.propTypes = {
    name: PropTypes.string
};

FirstApp.defaultProps = {
    name: 'Sin nombre especificado',
    greeting: 'Â¡Hola!'
}

export default FirstApp;