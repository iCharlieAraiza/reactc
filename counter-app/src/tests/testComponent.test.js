import '@testing-library/jest-dom/extend-expect'
import { fireEvent, render } from '@testing-library/react'
import FirstApp from '../FirstApp';
import React from 'react'


describe("First component test", ()=>{
    
    test("check content is render", ()=>{
        
        const note = {
            content: 'This is a test',
            important: true
        }

        const component = render(<FirstApp name="Carlos"/>)

        //console.log(component);
        component.getByText('Nombre: Carlos');

    })

    test("test click handle", ()=>{
        const mockHandler = jest.fn(); 
        const component = render(<FirstApp name="Carlos" mockHandler={ mockHandler }/>)

        const button = component.getByText('prueba de click');
        fireEvent.click( button );
        fireEvent.click( button );

        expect(mockHandler).toHaveBeenCalledTimes(1);
        
    })

});