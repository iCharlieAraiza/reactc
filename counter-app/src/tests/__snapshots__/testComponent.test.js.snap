// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `ShallowWrapper {}`;

exports[`Doing a component test first component test 1`] = `ShallowWrapper {}`;
